cmd/branchingconfigmanagers/README.md

```diff
@@ -0,0 +1,102 @@
+- Do not assume presence/absence of certain lifecycle phases. If the current phase of the product version is unknown to
+  the Config Manager, it should iterate the events to the past until it finds a known phase and enforce the state in
+  that phase.
+- Never modify git state. Config Managers should only change the CI configuration content. Committing the changes and
+  submitting Pull Requests for them is handled by a separate `prcreator` tool.
+- Never change any CI configuration other than the area the Config Manager "owns". Assume that all changes done,
+  including newly created files, will be committed to the repository.
+- If the Config Manager ends with zero exit code, all CI configuration must match the policy.
+- If the CI configuration does not match the policy after a Config Manager is executed, it must end with non-zero exit
+  code.
+- Avoid leaking confidential information. The OCP lifecycle data is provided to Config Managers via a
+  [ConfigMap](#configmap-with-ocp-lifecycle-data) because its content is confidential. Avoid logging dates and saving
+  artifacts that may contain them. Avoid even indirect information leaks, for example by doing actions like "10 days
+  before GA, enforce certain state": this leaks GA date 10 days in advance if Config Manager source code is available.
+
+### Config Managers Should
+
+- Avoid complex configuration as much as possible. Ideally, the Config Manager should have no configuration file at all.
+  Configuration file with an allow/denylist of repositories whose config is (not) managed is acceptable. Anything more
+  complicated should be avoided.
+- Be structured in a way that decisions about certain OCP version config are centralized. Sometimes these decisions are
+  not independent. For example, if your Config Manager needs to enforce certain state of the config for OCP version that
+  precedes the latest GA version, then the Config Manager should be structured as _"when processing a version X.Y,
+  determine which version is the latest GA, and if version X.Y is the preceding version, enforce certain state"_. It
+  should **not** be structured as _"when processing a version X.Y, determine if it is the GA version and if yes, enforce
+  certain state of the config belonging to the preceding version"_.
+- Change the CI state to conform the policy to the date and time of the execution, and allow to override this default
+  with a command line option.
+
+## Deploying New Config Managers
+
+Any new config manager should be deployed to run in a periodic Prow job which mounts a ConfigMap `TBD` that contains the
+YAML file with [OCP lifecycle data](#product-lifecycle-data). The Prowjob should first run the Config Manager, then if
+it ended with zero exit code, it should use the `prcreator` tool to commit the changes and submit a PR to propagate the
+changes.
+
+TODO: What period
+
+### ConfigMap with OCP Lifecycle Data
+
+TODO: What ConfigMap
```

Comment:
@stevekuznetsov do we know this information already?

Reply:
No, this is TODO

Reply:
I'll add a note about why this is a TODO and when it needs to be filled in, then I think we can merge, WDYT?. It's useful for CM authors even without discussing how to deploy.